### PR TITLE
DOC: update the DataFrame.iat[] docstring

### PR DESCRIPTION
**Pull request by akosel**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
# paste output of "scripts/validate_docstrings.py <your-function-or-method>" here
# between the "```" (remove this comment, but keep the "```")

################################################################################
####################### Docstring (pandas.DataFrame.iat) #######################
################################################################################

Selects a single value for a row/column pair by integer position.

Useful if performance is a major concern and you only need to
get or set a value at a particular row/column.

See Also
--------
at : Selects a single value for a row/column label pair
loc : Selects a group of rows and columns by label(s)
iloc : Selects group of rows and columns by integer position(s)

Returns
-------
    None

Examples
--------
>>> df = pd.DataFrame([[0,2,3], [0,4,1], [10,20,30]])
>>> df
    0   1   2
0   0   2   3
1   0   4   1
2  10  20  30
>>> df.iat[1, 2]
1
>>> df.iat[1, 2] = 10
>>> df.iat[1, 2]
10

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.iat" correct. :)

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -1920,11 +1920,35 @@ def _convert_key(self, key, is_setter=False):
...
+    Useful if performance is a major concern and you only need to
+    get or set a value at a particular row/column.
 
+    See Also
+    --------
+    at : Selects a single value for a row/column label pair
+    loc : Selects a group of rows and columns by label(s)
+    iloc : Selects group of rows and columns by integer position(s)
+
+    Returns
```
Comment (TomAugspurger) [inline pandas/core/indexing.py]: You can omit the returns  section if there's no return value

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -1920,11 +1920,35 @@ def _convert_key(self, key, is_setter=False):
...
+    -------
+        None
+
+    Examples
+    --------
+    >>> df = pd.DataFrame([[0,2,3], [0,4,1], [10,20,30]])
```
Comment (TomAugspurger) [inline pandas/core/indexing.py]: pep8: spaces after comma.

File: pandas/core/indexing.py [hunk 3]
```diff
@@ -1920,11 +1920,35 @@ def _convert_key(self, key, is_setter=False):
 
 
 class _iAtIndexer(_ScalarAccessIndexer):
-    """Fast integer location scalar accessor.
+    """
+    Selects a single value for a row/column pair by integer position.
```
Comment (jorisvandenbossche) [inline pandas/core/indexing.py]: Selects -> Select

File: pandas/core/indexing.py [hunk 4]
```diff
@@ -1920,11 +1920,35 @@ def _convert_key(self, key, is_setter=False):
...
 
-    Similarly to ``iloc``, ``iat`` provides **integer** based lookups.
-    You can also set using these indexers.
+    Useful if performance is a major concern and you only need to
+    get or set a value at a particular row/column.
 
+    See Also
+    --------
+    at : Selects a single value for a row/column label pair
```
Comment (jorisvandenbossche) [inline pandas/core/indexing.py]: I think we will need to do `DataFrame.at` to make the automatic link work (and the same for the others below)

File: pandas/core/indexing.py [hunk 5]
```diff
@@ -1920,11 +1920,35 @@ def _convert_key(self, key, is_setter=False):
 
 
 class _iAtIndexer(_ScalarAccessIndexer):
-    """Fast integer location scalar accessor.
+    """
+    Selects a single value for a row/column pair by integer position.
 
-    Similarly to ``iloc``, ``iat`` provides **integer** based lookups.
-    You can also set using these indexers.
+    Useful if performance is a major concern and you only need to
```
Comment (jorisvandenbossche) [inline pandas/core/indexing.py]: I would be a bit more careful with "major concern", as if performance is really important, you typically should not use this indexer many times. But, it is certainly useful to mention that `iat` is faster than `iloc` when you know you want to access only a single scalar.

File: pandas/core/indexing.py (hunk 3 above)
Comment (jorisvandenbossche) [reply pandas/core/indexing.py]: BTW, I think "Select" is not the correct term, as you can also use this to set values

File: pandas/core/indexing.py (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/core/indexing.py]: Also, can you specify string column names? eg `columns=['A', 'B', 'C']` 
to make it clear that it is position based (with the integer indexes it is dubious)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added some more comments

**Comment by jorisvandenbossche** [comment]
Thanks for the updates!

File: pandas/core/indexing.py [hunk 6]
```diff
@@ -1920,11 +1920,34 @@ def _convert_key(self, key, is_setter=False):
 
 
 class _iAtIndexer(_ScalarAccessIndexer):
-    """Fast integer location scalar accessor.
+    """
+    Access a single value for a row/column pair by integer position.
 
-    Similarly to ``iloc``, ``iat`` provides **integer** based lookups.
-    You can also set using these indexers.
+    Similar to ``iloc``, in that both provide integer-based lookups. Use
```
Comment (jreback) [inline pandas/core/indexing.py]: this also works on a Series, can you re-word to accomodate

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexing.py (hunk 6 above)
Comment (akosel) [reply pandas/core/indexing.py]: Great point, I specify that `iat` works for both a `DataFrame` and a `Series`. I also added an example for using it on `Series`.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks @akosel !

**Pull request merged**